src/main.cpp

```diff
@@ -116,14 +116,21 @@ namespace rhost {
 
     void set_memory_limit() {
 #ifdef WIN32
-        ULONGLONG total;
-        if (!GetPhysicallyInstalledSystemMemory(&total)) {
+        MEMORYSTATUSEX ms = {};
+        ms.dwLength = sizeof ms;
+        if (!GlobalMemoryStatusEx(&ms)) {
+            logf("Couldn't set R memory limit - GlobalMemoryStatusEx failed with GetLastError=%u\n", GetLastError());
             return;
         }
 
+        double memory_limit = static_cast<double>(ms.ullTotalPhys / 1024 / 1024);
```

Comment:
Yep, but should never actually be the case - I've yet to see a machine with physical memory not measured in whole megabytes :)

And even should it happen, rounding down is safe here. It's only a `double` because that's what R expects for this function (presumably to allow for bigger range, and because R doesn't have a 64-bit integer type). Internally, `in_memsize` will convert it back to bytes, and store it in a 64-bit int.